File: 1-twosum/src/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static void main(String[] args) {
        int[] nums = new int[]{2, 11, 7, 5};
        int target = 9;

        int[] result = new Solution().twoSum(nums, target);
        System.out.printf("%d, %d", result[0], result[1]);
    }

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int item = target - nums[i];
            if (map.containsKey(item)) {
                return new int[]{map.get(item), i};
            }
            map.put(nums[i], i);
        }
        return null;
    }
}

File: 1-twosum/src/Solution.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

from typing import List, Dict

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numOrderDict: Dict[int, int] = {}
        for index, num in enumerate(nums):
            numLeft = target - num
            if numLeft in numOrderDict:
                return [numOrderDict[numLeft], index]
            numOrderDict[num] = index
        return None

if __name__ == "__main__":
    nums: List[int] = [1, 2, 3, 5]
    target: int = 3
    result = Solution().twoSum(nums, target)
    if result is not None:
        print(result[0], result[1])
